### PR TITLE
Feature/Enhance Data Structure For LinktaFlow & UserInput

### LINKED ISSUE #328: Ensure Consistent LinktaFlow ID Handling and Move ID Mapping from Controller to Service
**Related Ticket**
#329 

**Description**
Currently, the ID mapping logic is handled within the controllers. This should be moved to the service layer to maintain a clean separation of concerns and ensure consistent ID mapping across the application.

Additionally, the LinktaFlow ID handling in the backend have inconsistencies, particularly in the ID mapping logic between different service methods. For example, the createLinktaFlow and fetchLinktaFlow methods have inconsistent ID mappings which can lead to bugs and maintenance challenges.

Examples of Inconsistencies:

Create LinktaFlow:
```typescript
const mappedNodes = nodes.map((node: CustomNode) => ({
  ...node._doc,
  id: node._id,
  _id: undefined,
}));

```
Fetch LinktaFlow:
```typescript
const mappedNodes = nodes.map((node: CustomNode) => ({
  ...node._doc,
  id: node.id,
}));
```

**Objective**
- Ensure consistent ID mapping logic within the backend services.
- Standardize the ID mapping logic in the service layer.

**Tasks**
Refactor Service:
- Move the ID mapping logic from the controllers to the corresponding services.
- Ensure the relevant service methods perform the mapping and return the mapped data.
- Ensure the ID mapping logic for linktaFlow is consistent in the applicaiton

Update the controllers:
- Remove the ID mapping logic from the controller.
- Ensure the controller calls the refactored service method and handles the response accordingly.

**Acceptance Criteria**
- ID mapping logic is handled within the service layer
- Controller fetches already mapped data from the service.
- Consistent ID mapping across the application.
- Proper error handling is in place for all operations.

### PR DESCRIPTION
**Pull request by kinskiwu**
## Description
Backend:
- [X] Simplify the LinktaFlow schema to include only essential fields & to optimize data structure
- [X] Remove redundant ID mapping
- [X] Relocate UserInput ID mapping from UserInputController to UserInputService - to adhere to the separation of concerns

Others
- [X] Update API design documentation
- [X] Re-organize type files - separate data model types into separate files

Todo
- [ ] Audit LinktaFlow zustand state on frontend
- [ ] Investigate why save LinktaFlow button not working
- [ ] Audit InputHistory data retrieval performance & decide whether to update the data models for User & UserInput

Fixes #328 #329

## Type of change
- [X] 📝 Documentation update
- [X] Other 🗒️ Enhancement

## How Can this be tested? Testing Plan to review this PR
- [X] Where in the app does this change take place?
- [X] What are the steps to test this change?
- [X] What is the expected output?
1. Spin up app at root directory
2. Click a `userInput` on the side nav
3. Click the `TensorFlow dev tool icon` and you will see the following updated data structures:
4. LinktaFlow:
image
5. InputHistory:
image

## Checklist:

- [X] I have performed a self-review of my own code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [X] I have made corresponding changes to the documentation.
- [X] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [X] I have reviewed the [**Security Best Practices Document**](./docs/SECURITY_BEST_PRACTICES.md).
- [X] I have followed the [**Naming Conventions Guide**](./docs/NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [X] I have ensured that my pull request title is descriptive.

### CODE REVIEW

File: server/controllers/userInputController.ts [hunk 1]
```diff
@@ -68,27 +72,10 @@ const createUserInputController = (
...
-          _id: undefined,
-        }));
-
-        res.locals.linktaFlow = {
-          id: _id,
-          userInputId,
-          nodes: mappedNodes,
-          edges: mappedEdges,
-        };
+        res.locals.linktaFlowId = newLinktaFlow._id;
```
Comment (kinskiwu) [inline server/controllers/userInputController.ts]: The reason for this change is becuase the response for generateLinktaFlowFromUserInput route is changed in PR #364

Comment (tedgusek) [reply server/controllers/userInputController.ts]: I just looked through #364 and approved that, this all looks great, thank you.

File: server/models/LinktaFlowModel.ts
Comment (tedgusek) [inline server/models/LinktaFlowModel.ts:1]: The code is so much cleaner now!

**Review by tedgusek — APPROVED** [review]
Very thorough work, and clear documentation.

File: server/controllers/userInputController.ts (hunk 1 above)
Comment (kinskiwu) [reply server/controllers/userInputController.ts]: Thank you Ted!

**Comment by colinvandergraaf** [comment]
I see you requested review on this, but it's still a draft PR. Is this ready to be reviewed?

**Comment by kinskiwu** [comment]
@colinvandergraaf Thanks for asking! This pr is not yet ready. I plan on making some changes in a bit. Will convert it to ready when I'm done.

**Comment by kinskiwu** [comment]
Closing this PR draft as I plan to revisit the design for all data entities as a whole first.

**Pull request closed without merging**